cloud/linode/loadbalancers.go

```diff
@@ -311,6 +305,33 @@ func (l *loadbalancers) updateNodeBalancer(
 				break
 			}
 		}
+		oldNBNodeIDs := make(map[string]int)
+		if currentNBCfg != nil {
+			// Obtain list of current NB nodes and convert it to map of node IDs
+			currentNBNodes, err := l.client.ListNodeBalancerNodes(ctx, nb.ID, currentNBCfg.ID, nil)
+			if err != nil {
+				klog.Warningf("Unable to list existing nodebalancer nodes for NB %d config %d, error: %s", nb.ID, newNBCfg.ID, err)
```

Comment:
Why don't we err out? Seems weird to ignore these nodes if we fail to list them.

Reply:
This is because the issue with this specific call does not prevent us succeeding with the configuration even it is suboptimal, but if we bail out, it would malfunction completely.